evm_body.yaml

```diff
@@ -1578,6 +1576,50 @@ address_and_blockNumber_param_eth:
       maxItems: 2
       items:
         type: string
+
+BlockNumberOrTagOrHash:
+  title: Block number or tag, or block hash
+  oneOf:
+    - title: Block Number
+      $ref: '#/BlockNumber'
+    - title: Block Tag
+      $ref: '#/BlockTag'
+    - title: Block Hash
+      $ref: '#/BlockHash'
+
+BlockTag:
+  title: Block tag
+  type: string
+  description: |
+    One of the following **block tags**:
+      * `pending` - A sample next block built by the client on top of latest and containing the set of transactions usually taken from local mempool. Intuitively, you can think of these as blocks that have not been mined yet.
+      * `latest` - The most recent block in the canonical chain observed by the client, this block may be re-orged out of the canonical chain even under healthy/normal conditions.
+      * `earliest` - The lowest numbered block the client has available. Intuitively, you can think of this as the first block created.
+  enum:
+    - pending
+    - latest
+    - earliest
+  default: latest
+
+BlockNumber:
+  title: Block number
+  type: string
+  description: The hex value of a **block number**.
+
+BlockHash:
+  title: Block hash
+  type: object
+  required:
+    - blockHash
+  properties:
+    blockHash:
+      title: Block hash
+      type: string
+    requireCanonical:
+      title: Require canonical
+      type: boolean
+      default: false
```

Comment:
would be awesome to have a default value here - we can use one that works for ethereum (doesn't have to work for all chains bc we can override on those docs)

Reply:
hmm actually should we re-use the generic blockHash component that we use in other methods ex: eth_getBlockByHash? This way we can also pull in the same types and descriptions